sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -731,6 +731,91 @@ class RocksDB(
     this
   }
 
+  /**
+   * Create an empty RocksDB state store at the specified version without loading previous data.
+   *
+   * This method is useful when state will be completely rewritten and
+   * does not need to load previous states
+   *
+   * @param targetVersion The version to initialize the empty store at (must be >= 0)
+   * @param stateStoreCkptId Optional checkpoint ID (required if checkpoint IDs are enabled)
+   * @param readOnly Whether to open the store in read-only mode
+   * @return A RocksDB instance with an empty state at the target version
+   */
+  def loadEmpty(
```

Comment:
We shouldn't use a separate function. Lets implement this within the normal load func to avoid duplicating code.

You can implement for only `loadWithoutCheckpointId` for now. We will do for `loadWithCheckpointId` when we add checkpoint v2 support.